scaleway/resource_domain_record.go

```diff
@@ -0,0 +1,425 @@
+func resourceScalewayDomainRecordRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	domainAPI := domainAPI(meta)
+	var record *domain.Record
+
+	if strings.Contains(d.Id(), "/") {
+		tab := strings.SplitN(d.Id(), "/", -1)
+		if len(tab) != 2 {
+			return diag.FromErr(fmt.Errorf("cant parse record id: %s", d.Id()))
+		}
+
+		dnsZone := tab[0]
+		id := tab[1]
+
+		res, err := domainAPI.ListDNSZoneRecords(&domain.ListDNSZoneRecordsRequest{
+			DNSZone: dnsZone,
+		}, scw.WithAllPages())
+
+		if err != nil {
+			if is404Error(err) {
+				d.SetId("")
+				return nil
+			}
+			return diag.FromErr(err)
+		}
+
+		for _, r := range res.Records {
+			if r.ID == id {
+				record = r
+				break
+			}
+		}
+	} else {
+		res, err := domainAPI.ListDNSZoneRecords(&domain.ListDNSZoneRecordsRequest{
+			DNSZone: d.Get("dns_zone").(string),
+			Name:    d.Get("name").(string),
+			Type:    domain.RecordType(d.Get("type").(string)),
+		}, scw.WithAllPages())
+
+		if err != nil {
+			if is404Error(err) {
+				d.SetId("")
+				return nil
+			}
+			return diag.FromErr(err)
+		}
+
+		for _, r := range res.Records {
+			if flattenDomainData(r.Data, r.Type).(string) == d.Get("data").(string) {
+				record = r
+				break
+			}
+		}
+	}
+
+	if record == nil {
+		d.SetId("")
+		return nil
+	}
+
+	d.SetId(record.ID)
+	_ = d.Set("data", flattenDomainData(record.Data, record.Type))
+	_ = d.Set("ttl", int(record.TTL))
+	_ = d.Set("priority", int(record.Priority))
+	_ = d.Set("geo_ip", flattenDomainGeoIP(record.GeoIPConfig))
+	_ = d.Set("http_service", flattenDomainHTTPService(record.HTTPServiceConfig))
+	_ = d.Set("weighted", flattenDomainWeighted(record.WeightedConfig))
+	_ = d.Set("view", flattenDomainView(record.ViewConfig))
+
+	return nil
+}
+
+func resourceScalewayDomainRecordUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	domainAPI := domainAPI(meta)
+
+	if d.HasChangesExcept("dns_zone", "keep_empty_zone", "name", "type") {
+		id := d.Id()
+
+		geoIP, okGeoIP := d.GetOk("geo_ip")
+		_, err := domainAPI.UpdateDNSZoneRecords(&domain.UpdateDNSZoneRecordsRequest{
+			DNSZone: d.Get("dns_zone").(string),
+			Changes: []*domain.RecordChange{
+				{
+					Set: &domain.RecordChangeSet{
+						ID: &id,
+						Records: []*domain.Record{
+							{
+								Name:              d.Get("name").(string),
+								Data:              d.Get("data").(string),
+								Priority:          uint32(d.Get("priority").(int)),
+								TTL:               uint32(d.Get("ttl").(int)),
+								Type:              domain.RecordType(d.Get("type").(string)),
+								GeoIPConfig:       expandDomainGeoIPConfig(d.Get("data").(string), geoIP, okGeoIP),
+								HTTPServiceConfig: expandDomainHTTPService(d.GetOk("http_service")),
+								WeightedConfig:    expandDomainWeighted(d.GetOk("weighted")),
+								ViewConfig:        expandDomainView(d.GetOk("view")),
+							},
+						},
+					},
+				},
+			},
+			ReturnAllRecords: nil,
```

Comment:
Why not telling explicelty whether we want to return all records or not? Should we put a `scw.BoolPtr(false)` ?